system-addon/common/Reducers.jsm

```diff
@@ -207,6 +207,18 @@ function Sections(prevState = INITIAL_STATE.Sections, action) {
           // If the action is updating rows, we should consider initialized to be true.
           // This can be overridden if initialized is defined in the action.data
           const initialized = action.data.rows ? {initialized: true} : {};
+
+          if (action.data.rows && section.rows.find(card => card.pinned)) {
```

Comment:
Why the additional `find` check? Doesn't seem necessary?

Reply:
Just so we don't copy the rows, and add the additional processing (iterating over all rows), if there's no pinned card (almost always).

Reply:
Eh. I don't think this is particularly performance critical section, so simpler code is preferable, but this is fine.